### DISCUSSION TITLE
Implement AMD FidelityFX Brixelizer GI in the Vulkan renderer

### DISCUSSION THREAD

**Discussion by mahdisml** [godotengine/godot-proposals#10180] [3D]
### Describe the project you are working on

3D horror game

### Describe the problem or limitation you are having in your project

Dynamic GI

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

AMD FidelityFX Brixelizer + Brixelizer GI features the following:

Denoised indirect diffuse and indirect specular outputs
Implemented in compute without hardware-accelerated ray-tracing
Optional modes to output at lower than native resolutions
Native DirectX® 12 and Vulkan® SDK backend implementation libraries
Fully dynamic global illumination with multiple bounces

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

https://gpuopen.com/learn/getting-the-most-out-of-fidelityfx-brixelizer/
https://gpuopen.com/fidelityfx-brixelizer/

### If this enhancement will not be used often, can it be worked around with a few lines of script?

idk

### Is there a reason why this should be core and not an add-on in the asset library?

HDDAGI vs SDFGI vs Brixelizer GI
It's a complicated issue :) I have no expertise in this field

**Comment by Calinou** [comment]
Moving to discussion, as there is no concrete technical implementation provided.

Note that I don't expect Brixelizer to be useful in Godot, since [HDDAGI](https://github.com/godotengine/godot/pull/86267) is nearing completion and will be finished for a future 4.x release. There isn't much that is missing in HDDAGI, so I would be surprised if it was scrapped and replaced with a different solution entirely.

I expect future GI efforts to be more focused on raytracing (such as [GI-1.1](https://github.com/GPUOpen-LibrariesAndSDKs/Capsaicin)) once HDDAGI is working well. These techniques can provide higher visual quality without the limitations of SDF-based approaches, such as slow (re)generation times which can be the source of traversal stutter. I'd say this is 2025+ territory either way, at which point support for raytracing will start to become more widespread.

**Reply by NatanAmorim** [reply]
> I would be surprised if it was scrapped and replaced with a different solution entirely.

well... Jamsers did [comment](https://github.com/godotengine/godot/pull/86267#issuecomment-2538654729) on [Upgrade SDFGI to HDDAGI #86267](https://github.com/godotengine/godot/pull/86267).

> ...He's (Juan) apparently thought of an approach that's even better than HDDAGI, and will eventually work on that instead of continuing HDDAGI...

Edit: Added more context.

**Reply by VicPrzz00** [reply]
too bad that even a year later, it turns out that is has been scrapped even though it was apparently almost finished. it still hasnt been implemented as an option, be it experimental.

**Reply by NatanAmorim** [reply]
it was experimental at some point before 4.3 came out ([Godot Has A New Global Illumination System ...Again](https://www.youtube.com/watch?v=3vVx90myfyU)), but "almost finished" can be deceiving, the 80/20 rule comes to mind.

**Reply by NuclearPhoenixx** [reply]
At the very least I would've hoped them to use this as a temporary fix, but instead we're now stuck even longer with SDFGI. It would've been very interesting to see, and I hope we'll have a real useful GI system in the future. I've somewhat lost hope to be honest :/

**Comment by mahdisml** [comment]
Hey @reduz ,
Loving all the progress on the new GI systems, super excited for HDDAGI!
Just wanted to put "Brixelizer GI" on your radar from a technical standpoint. Its approach to voxel octrees seems really clever.
No pressure to dive deep, just thought it was a cool piece of tech worth sharing. Keep up the amazing work! ❤️

**Comment by luzidd** [comment]
This brixelizer looks really interesting. Is anyone working on this?
